rayon-demo doesn't compile with Rust nightly

```
   Compiling rayon-demo v0.0.0 (file:///home/markus/rayon/rayon-demo)
error[E0482]: lifetime of return value does not outlive the function call
  --> src/tsp/graph.rs:50:49
   |
50 |     pub fn edges<'a>(&'a self, source: Node) -> impl Iterator<Item=Edge> {
   |                                                 ^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the return value is only valid for the lifetime 'a as defined on the method body at 50:5
  --> src/tsp/graph.rs:50:5
   |
50 | /     pub fn edges<'a>(&'a self, source: Node) -> impl Iterator<Item=Edge> {
51 | |         self.all_nodes()
52 | |             .filter_map(move |target| {
53 | |                 self.edge_weight(source, target)
...  |
61 | |             })
62 | |     }
   | |_____^

error: aborting due to previous error

error: Could not compile `rayon-demo`.

To learn more, run the command again with --verbose.
```

--- a/rayon-demo/src/tsp/graph.rs
+++ b/rayon-demo/src/tsp/graph.rs
@@ -47,7 +47,7 @@ impl Graph {
         if w.is_max() { None } else { Some(w) }
     }
 
-    pub fn edges<'a>(&'a self, source: Node) -> impl Iterator<Item=Edge> {
+    pub fn edges<'a>(&'a self, source: Node) -> impl Iterator<Item=Edge> + 'a {
         self.all_nodes()
             .filter_map(move |target| {
                 self.edge_weight(source, target)

